app/services/storage.js

```diff
@@ -375,10 +375,32 @@ export class ProfileStorage {
     return this.collectURLs(await this.db.all(query, [since, limit]));
   }
 
-  async starred(): Promise<[string]> {
+  async getStarredWithOrderByAndLimit(orderClause: string, limit: ?number): Promise<[string]> {
```

Comment:
Wow, this is rough.  The caller needs to know the join involves `s` and `t`.  Maybe just special case ordering as `newestFirst: boolean`.